src/tools/apicall/apicall.ts

```diff
@@ -0,0 +1,131 @@
+  description: string;
+  openApiSchema: any;
+  apiKey?: string;
+  http_proxy_url?: string;
+
+  inputSchema() {
+    return {
+      type: 'object',
+      required: ['path', 'method'],
+      oneOf: Object.entries(this.openApiSchema.paths).flatMap(([path, pathSpec]: [string, any]) =>
+        Object.entries(pathSpec).map(([method, methodSpec]: [string, any]) => ({
+          additionalProperties: false,
+          properties: {
+            path: {
+              const: path,
+              description:
+                'Do not replace variables in path, instead of, put them to the parameters object.'
+            },
+            method: { const: method, description: methodSpec.summary || methodSpec.description },
+            ...(methodSpec.requestBody?.content?.['application/json']?.schema
+              ? {
+                  body: methodSpec.requestBody?.content?.['application/json']?.schema
+                }
+              : {}),
+            ...(methodSpec.parameters
+              ? {
+                  parameters: {
+                    type: 'object',
+                    additionalProperties: false,
+                    required: methodSpec.parameters
+                      .filter((p: any) => p.required === true)
+                      .map((p: any) => p.name),
+                    properties: methodSpec.parameters.reduce(
+                      (acc: any, p: any) => ({
+                        ...acc,
+                        [p.name]: { ...p.schema, description: p.name }
+                      }),
+                      {}
+                    )
+                  }
+                }
+              : {})
+          }
+        }))
+      )
+    } as const satisfies SchemaObject;
+  }
+
+  public constructor({ name, description, openApiSchema, apiKey, http_proxy_url, ...rest }: ApiCallToolOptions) {
+    super({ name, description, ...rest });
+    this.name = name;
+    this.apiKey = apiKey;
+    this.http_proxy_url = http_proxy_url;
+    this.description = description ?? 'Use input schema to infer description';
+    this.openApiSchema = parse(openApiSchema);
+    // TODO: #105 review error codes. Were using APIErrorCode, now ToolError
+    if (!this.openApiSchema?.paths) {
+      throw new ToolError("Server is not specified!")
+    }
+  }
+
+  protected async _run(input: any) {
+    let path: string = input.path || '';
+    const url = new URL(this.openApiSchema.servers[0].url);
+    Object.keys(input.parameters ?? {}).forEach((key) => {
+      if (path.search(`{${key}}`) >= 0) {
+        path = path.replace(`{${key}}`, input.parameters[key]);
+      } else {
+        url.searchParams.append(key, input.parameters[key]);
+      }
+    });
+    url.pathname = join(url.pathname, path);
+    const headers: { [key: string]: string } = { Accept: 'application/json' };
+    if (this.apiKey) {
+      //TODO: #105 confirm ok to remove use of encrypted api key
+      headers['Authorization'] = `Bearer ${this.apiKey}`;
+    }
```

Comment:
I would consider refactoring the whole URL building part by leveraging `createURLParams` function from  `src/internals/fetcher.ts` module.